src/core/reference/include/openvino/reference/autobroadcast_binop.hpp

```diff
@@ -99,203 +320,39 @@ void autobroadcast_binop(const T* arg0,
-                const auto arg1_coord = util::reduce(output_coord, arg1_squeezed_axes);
-                const auto out_index = coordinate_index(output_coord, arg0_shape);
-                const auto arg0_index = coordinate_index(output_coord, arg0_shape);
-                const auto arg1_index = coordinate_index(arg1_coord, arg1_squeezed_shape);
-                out[out_index] = elementwise_functor(arg0[arg0_index], arg1[arg1_index]);
-            }
-        }
+        pdpd_broadcast_binop(arg0, arg1, out, arg0_shape, arg1_shape, broadcast_spec.m_axis, elementwise_functor);
+        break;
+    default:
+        break;
     }
 }
 
-/// \brief Helper function to implement autobroadcasting elementwise ternaryop
-/// references.
-///
-/// \tparam U Element type of the selector tensor.
-/// \tparam T Element type of the input tensors.
-/// \tparam Functor Type of the functor for the elementwise operation. Must support
-///                 operator()(U,T,T), and operator()(U,T,T) must return a value of type
-///                 T.
-///
-/// \param arg0 Pointer to the buffer for selector tensor.
-/// \param arg1 Pointer to the buffer for left operand input tensor.
-/// \param arg2 Pointer to the buffer for right operand input tensor.
-/// \param out Pointer to the buffer for output tensor. This must be pre-allocated by
-///            the caller, and must be large enough to hold a tensor of the correct
-///            shape.
-/// \param broadcast_spec Specification of the auto-broadcasting scheme.
-/// \param elementwise_functor Functor implementing the elementwise operation to be
-///                            applied across the input tensors. Must accept an argument
-///                            of
-///                            type U and two of type T, and return a value of type T.
+/**
+ *
+ * \brief Helper function to implement auto broadcasting elementwise ternary op references.
+ * \tparam U Element type of the selector tensor.
+ * \tparam T Element type of the input tensors.
+ * \tparam Functor Type of the functor for the elementwise operation. Must support
+ *                 operator()(U,T,T), and operator()(U,T,T) must return a value of type
+ *                 T.
+ * \param arg0 Pointer to the buffer for selector tensor.
+ * \param arg1 Pointer to the buffer for left operand input tensor.
+ * \param arg2 Pointer to the buffer for right operand input tensor.
+ * \param out Pointer to the buffer for output tensor. This must be pre-allocated by
+ *            the caller, and must be large enough to hold a tensor of the correct
+ *            shape.
+ * \param broadcast_spec Specification of the auto-broadcasting scheme.
+ * \param elementwise_functor Functor implementing the elementwise operation to be
+ *                            applied across the input tensors. Must accept an argument
+ *                            of
+ *                            type U and two of type T, and return a value of type T.
+ */
```

Comment:
It looks like cut to 80 columns. Would it look better cut to 120? I hope so ;)

Reply:
Done